nixarr/prowlarr/default.nix

```diff
@@ -84,40 +84,117 @@ in {
       "d '${cfg.stateDir}' 0700 ${globals.prowlarr.user} root - -"
     ];
 
+    networking.firewall = mkIf cfg.openFirewall {
+      allowedTCPPorts = [cfg.port];
+    };
+
+    users = {
+      groups.${globals.prowlarr.group}.gid = globals.gids.${globals.prowlarr.group};
+      users.${globals.prowlarr.user} = {
+        isSystemUser = true;
+        group = globals.prowlarr.group;
+        uid = globals.uids.${globals.prowlarr.user};
+      };
+    };
     systemd.services.prowlarr = {
       description = "prowlarr";
       after = ["network.target"];
       wantedBy = ["multi-user.target"];
       environment.PROWLARR__SERVER__PORT = builtins.toString cfg.port;
 
+      postStart = mkIf nixarr.autosync (
+        let
+          configure-prowlarr =
+            pkgs.writers.writePython3Bin "configure-prowlarr" {
+              libraries = [];
+              flakeIgnore = ["E501" "E121" "E122"];
+            } ''
+              import sqlite3
+              import json
+              import os.path
+              import time
+
+              db_path = "${cfg.stateDir}/prowlarr.db"
+              while not os.path.exists(db_path):
+                  time.sleep(1)
+
+              con = sqlite3.connect(db_path)
+              api_key = open("${nixarr.stateDir}/api-key", "r").read()
+              sonarr = {
+                "prowlarrUrl": "http://localhost:9696",
+                "baseUrl": "http://localhost:8989",
+                "apiKey": api_key,
+                "syncCategories": [
```

Comment:
i'm not super deep in the *arr ecosystem, but this looks like a candidate for a module option?